lib/worker_manager.go

```diff
@@ -311,7 +387,7 @@ func (m *workerManagerImpl) HandleHeartbeat(msg *HeartbeatPingMessage, fromNode
 	log.L().Debug("received heartbeat", zap.Any("msg", msg))
 	workerInfo, ok := m.workerInfos[msg.FromWorkerID]
 	if !ok {
-		if m.initialized {
+		if m.fsmState.Load() != workerManagerWaitingHeartbeats {
```

Comment:
If the `fsmState` has been `workerManagerNormal`, and a worker with the same id restarts, will the worker id in `m.workerInfos`?

Reply:
In the case, the restart must have been triggered by `CreateWorker` on this master, so the worker info is added already.